### PR TITLE
Fix #5612 phantomjs issue

### LINKED ISSUE #5612: Recent removal of postinstall of phantomjs breaks karma tests
Recent removal of postinstall of phantomjs breaks karma tests. Ref: https://github.com/jhipster/generator-jhipster/issues/5553 cc @ruddell 

##### **Motivation for or Use Case**

I was experiencing build errors in jenkins but can reproduce locally as well.

##### **Reproduce the error**

yo jhipster (using 4.3.0 and .yo-rc.json below)
./mvnw -Pprod package -DskipTests

=>ERROR [plugin]: Error during loading "/Users/sdoxsee/workspace/test/node_modules/karma-phantomjs-launcher" plugin:
[INFO]   Path must be a string. Received null

Then hangs

##### **Related issues**

https://github.com/jhipster/generator-jhipster/issues/5553

##### **Suggest a Fix**

Re-add the postinstall of phantomjs to package.json?

##### **JHipster Version(s)**

```
test@0.0.0 /Users/sdoxsee/workspace/test
├── UNMET PEER DEPENDENCY @angular/common@4.0.0
├── UNMET PEER DEPENDENCY @angular/core@4.0.0
└── generator-jhipster@4.3.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

```yaml
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "4.3.0",
    "baseName": "test",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "3b6b84b614d0e6b3e618e3af72a19c2ac477a3a8",
    "clientFramework": "angular2",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

ls: no such file or directory: .jhipster/*.json

##### **Browsers and Operating System**

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

git version 2.12.1

node: v7.7.4

npm: 4.1.2

yeoman: 1.8.5

yarn: 0.22.0

Docker version 17.03.1-ce, build c6d412e

docker-compose version 1.11.2, build dfed245
'''

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

N/A

##### **Browsers and Operating System**

MacOS Sierra 10.12.4
Browser: N/A

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by sdoxsee**
- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/angular/_package.json [hunk 1]
```diff
@@ -151,7 +151,7 @@
     "e2e": "protractor <%= TEST_SRC_DIR %>protractor.conf.js",
     "postinstall": "webdriver-manager update && <%= clientPackageManager %> run webpack:build"
     <%_ } else { _%>
-    "postinstall": "<%= clientPackageManager %> run webpack:build"
+    "postinstall": "node node_modules/pngquant-bin/lib/install.js && <%= clientPackageManager %> run webpack:build"
```
Comment (pascalgrimaud) [inline generators/client/templates/angular/_package.json]: shouldn't it be `node node_modules/phantomjs-prebuilt/install.js &&` instead ?

File: generators/client/templates/angular/_package.json [hunk 2]
```diff
@@ -151,7 +151,7 @@
     "e2e": "protractor <%= TEST_SRC_DIR %>protractor.conf.js",
     "postinstall": "webdriver-manager update && <%= clientPackageManager %> run webpack:build"
```
Comment (pascalgrimaud) [inline generators/client/templates/angular/_package.json]: I know it was missing here, but I'm pretty sure it should be added here too

File: generators/client/templates/angular/_package.json (hunk 1 above)
Comment (ruddell) [inline generators/client/templates/angular/_package.json]: We need the phantomjs part (not the pngquant part which was removed)

File: generators/client/templates/angular/_package.json (hunk 2 above)
Comment (ruddell) [inline generators/client/templates/angular/_package.json]: Could you also add it in the other postinstall step too please

File: generators/client/templates/angular/_package.json (hunk 1 above)
Comment (sdoxsee) [inline generators/client/templates/angular/_package.json]: absolutely. my bad. will fix

Comment (sdoxsee) [inline generators/client/templates/angular/_package.json]: absolutely. my bad. will fix

**Pull request closed without merging**